doc/api/fs.md

```diff
@@ -1308,6 +1308,37 @@ characters directly to the `prefix` string. For instance, given a directory
 `prefix` must end with a trailing platform-specific path separator
 (`require('node:path').sep`).
 
+### `fsPromises.mkdtempDisposable(prefix[, options])`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `prefix` {string|Buffer|URL}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+* Returns: {Promise} Fulfills with a Promise for an async-disposable Object:
+  * `path` {string} The path of the created directory.
```

Comment:
If the prefix is a `Buffer`, then the resulting `path` here should be as well, otherwise there's a risk of the `path` being incorrect if the underlying path is not actually UTF8 encoded. That said, this might be a general problem with the existing API. Might need to be looked into separately.